packages/ui/src/components/SourceCode/SourceCode.tsx

```diff
@@ -56,8 +58,20 @@ export const SourceCode: FunctionComponent<SourceCodeProps> = (props) => {
     quickSuggestions: { other: true, strings: true, comments: true },
   });
 
+  const undoAction = () => {
+    (editorRef.current?.getModel() as any)?.undo();
+  }
+
+  const redoAction = () => {
+    (editorRef.current?.getModel() as any)?.redo();
+  }
+
   const customControls = useMemo(() => {
-    return [<SyncButton key="sync-button" isDirty={false} isVisible onClick={() => undefined} />];
+    return [
+      <UndoButton key="undo-button" isVisible onClick={() => undoAction} />,
+      <RedoButton key="redo-button" isVisible onClick={() => redoAction} />,
+      <SyncButton key="sync-button" isDirty={false} isVisible onClick={() => undefined} />,
```

Comment:
We added this sync button just in case, I think we can remove it as it's not doing anything

Reply:
ok